cmake/Modules/Testing.cmake

```diff
@@ -16,6 +16,18 @@ if(ENABLE_TESTING)
   set(MEMORYCHECK_COMMAND "${VALGRIND_BINARY}" CACHE FILEPATH "Memory Check Command")
   set(MEMORYCHECK_COMMAND_OPTIONS "${VALGRIND_DEFAULT_OPTIONS}" CACHE STRING "Memory Check Command Options")
 
+  # check if a faster linker is available
+  include(CheckCXXCompilerFlag)
+  check_cxx_compiler_flag(-fuse-ld=lld HAVE_LLD_LINKER)
+  check_cxx_compiler_flag(-fuse-ld=gold HAVE_GOLD_LINKER)
+  if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.13)
```

Comment:
If it is only for CMake>=3.13, this `if` should go on the outside. And maybe only do it for `ENABLE_TESTING=ON`.

Reply:
Yes, the if can be moved. But on the other hand, I submitted the PR to get some comments from @rbberger on whether he considers this kind of change useful for integration testing and in general. Then support for CMake 3.10-3.12 could be added as an else branch and the whole block moved to the toplevel CMakeList.txt file.

The entire file is already bracketed by a check of `ENABLE_TESTING`. That is why I put the check here in the first place.